server/lib/utils/util.go

```diff
@@ -50,3 +53,48 @@ func IFIP(name string, need_ipv6 bool) (net.IP, error) {
 	}
 	return nil, fmt.Errorf("interface %v: no suitable setting found", name)
 }
+
+type WaitGroup struct {
+	sync.WaitGroup
+	errors []error
+	cnt int32
+}
+
+func NewWaitGroup(size int) *WaitGroup {
+	wg := &WaitGroup {
+		errors: make([]error, size),
+	}
+	wg.WaitGroup.Add(size)
+	return wg
+}
+
+func (wg *WaitGroup) Done(err error) {
+	cnt := atomic.AddInt32(&wg.cnt, 1)
+	//log.Printf("Done with %v %v", cnt, err)
+	wg.errors[cnt - 1] = err
+	wg.WaitGroup.Done()
+}
+
+func (wg *WaitGroup) Wait() error {
+	wg.WaitGroup.Wait()
+	var err error
+	for i, e := range wg.errors {
+		if e != nil && err == nil {
+			err = e
+		}
+		wg.errors[i] = nil
+	}
+	wg.cnt = 0
+	//log.Printf("Wait with %v", err)
+	return err
+}
+
+//dump system stats
```

Comment:
__[gometalinter]__
* [golint] comment on exported function DumpMemUsage should be of the form "DumpMemUsage ..."